be/src/Unic.UrlMapper2/code/Services/RedirectSearcher.cs

```diff
@@ -108,7 +123,7 @@ protected virtual IQueryable<RedirectSearchResultItem> GetSearchQuery(IProviderS
             // r.RegexEnabled && redirectSearchData.SourceTerm.Match(r.SourceTerm)
             // However, this seems to generate some really weird SolR queries, resulting in no matches.
             // Therefore we have to load all entries being a regex enabled redirect and then do the filtering in memory.
-            return r => !r.RegexEnabled && r.SourceTerm == redirectSearchData.SourceTerm || r.RegexEnabled;
+            return r => (!r.RegexEnabled && !r.PreserveQueryString && r.SourceTerm == redirectSearchData.SourceTerm) || r.RegexEnabled || r.PreserveQueryString;
```

Comment:
I don't understand the need for this change, when `PreserveQueryString` is `true`, `RegexEnabled ` is `true` as well, right?

Reply:
When `PreserveQueryString` is `true` I also have to set `RegexEnabled` to `true` in code (so yes, you have right) but this is done later, in the `MapToSearchResult` method. So in here I want to take results either when `SourceTerm` matches the one from url AND `RegexEnabled` is `false` AND `PreserveQueryString` is `false`, or I'm taking all results where `RegexEnabled` is `true` or I need to consider situations when the `PreserveQueryString` is `true` but `RegexEnabled` might be set `false`. The best way I guess from user perspective also would be to have conditional rule between sitecore template fields, when user checks `PreserveQueryString` checkbox, then the `RegexEnabled` should be automatically checked as well. I invested a bit of time to setup that rule or at least the validation warning but it didn't work ;-) I guess it would need some event/processor in pipeline.
Summarizing, when the `PreserveQueryString` is checked, the code logic will work in the same way no matter if the `RegexEnabled` is `true` or `false`